### PR TITLE
[Jormun] Improve Authentification

### PR DESCRIPTION
**Pull request by AzHicham**
- Improve generated SqlAlchemy request (noload of linked tables, ie with relationships) 
- Add more logs

In order to improve sql requests generated by SQLAlchemy, we changed the Default Relationship Loading Technique ie 'joined' by 'noload'. 
Example : We have two tables : User & Jobs , Table has a foreign key job_id that match column id of Jobs table. 
- using mode 'joined' : when we request the table User we get also data in Job table 
- using mode 'noload' : when we request the table User we get only data from User table 

For more information please see the following documentation 
https://docs.sqlalchemy.org/en/14/orm/loading_relationships.html

### CODE REVIEW

File: source/jormungandr/jormungandr/authentication.py [hunk 1]
```diff
@@ -144,7 +149,8 @@ def has_access(region, api, abort, user):
         # To manage database error of the following type we should fetch one more time from database
         # Can connect to database but at least one table/attribute is not accessible due to transaction problem
         if can_read_user():
-            abort_request(user=user)
+            context = 'User is undefined, but users are accessible in database'
```
Comment (woshilapin) [inline source/jormungandr/jormungandr/authentication.py]: I'm not sure that I understand what this message means. Does the user exist or not?

File: source/jormungandr/jormungandr/instance_manager.py [hunk 2]
```diff
@@ -349,7 +350,8 @@ def get_instances(self, name=None, lon=None, lat=None, object_id=None, api='ALL'
         valid_instances = self._filter_authorized_instances(available_instances, api)
         if available_instances and not valid_instances:
             # user doesn't have access to any of the instances
-            authentication.abort_request(user=authentication.get_user())
+            context = 'user does not have access to any of the instances'
```
Comment (woshilapin) [inline source/jormungandr/jormungandr/instance_manager.py]:
```suggestion
            context = 'User does not have access to any of the instances'
```

(only for homogeneity with the rest).

**Review by woshilapin — APPROVED** [review]
Do we have DEBUG logs in production?

**Comment by AzHicham** [comment]
> Do we have DEBUG logs in production?

Unfortunately yes

File: source/jormungandr/jormungandr/utils.py [hunk 3]
```diff
@@ -689,6 +689,7 @@ def can_connect_to_database():
         connection.close()
         g.can_connect_to_database = True
     except Exception:
+        logging.getLogger(__name__).error('Connection to database has failed')
```
Comment (xlqian) [inline source/jormungandr/jormungandr/utils.py]:
```suggestion
        logging.getLogger(__name__).exception('Connection to database has failed')
```

**Review by kadhikari — APPROVED** [review]

**Review by azime — APPROVED** [review]

**Review by xlqian — APPROVED** [review]

**Pull request merged**